torch/csrc/autograd/python_cpp_function.h

```diff
@@ -22,7 +22,7 @@ PyObject* CppFunction_pynew(PyTypeObject *type, PyObject *args, PyObject *kwds)
   if (!obj) return NULL;
   THPCppFunction* f = (THPCppFunction*)obj.get();
   HANDLE_TH_ERRORS
-  new (&f->cdata) std::shared_ptr<Function>(Ctor()(args));
+  f->cdata.reset(Ctor()(args));
```

Comment:
The previous code called the C++ `shared_ptr<Function>` constructor, which initializes the object (but doesn't allocate memory because of the `(&f->cdata)` syntax).

The new code never calls the constructor. It probably doesn't have an effect here, since:
a) the constructor most likely just zero-initializes some fields
b) `tp_alloc` already zeros out the entire containing struct

But it's probably not good to assume (a).